src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -244,17 +323,22 @@ public override int GetHashCode(string obj)
                 throw new ArgumentNullException(nameof(obj));
 #endif
             }
-            return obj.GetHashCode();
+            return TextInfo.GetHashCodeOrdinalIgnoreCase(obj);
         }
 
         // Equals/GetHashCode methods for the comparer itself. 
-        public override bool Equals(object obj) => obj is OrdinalComparer;
-        public override int GetHashCode() => nameof(OrdinalComparer).GetHashCode();
+        public override bool Equals(object obj) => obj is OrdinalCaseSensitiveComparer;
```

Comment:
This should just use the parent implementation, not override.

The current impl is not symmetric: OrdinalComparer.Equals(OrdinalCaseSensitiveComparer) is true;  OrdinalCaseSensitiveComparer.Equals(OrdinalComparer) is false.